execution.bs

```diff
@@ -3173,67 +3196,91 @@ Insert this section as a new subclause between Header `<stop_token>` synopsis <b
 
 <ins>
 <blockquote>
-1. The `stoppable_token` concept checks for the basic interface of a “stop token” which is copyable and allows polling to see if stop has been requested and also whether a stop request is possible. It also requires an associated nested template-type-alias,
-    `T::callback_type<CB>`, that identifies the stop-callback type to use to register a callback to be executed if a stop-request is ever made on a stoppable_token of type, `T`. The `stoppable_token_for` concept checks for a stop token type compatible with a given
-    callback type. The `unstoppable_token` concept checks for a stop token type that does not allow stopping.
+1. The `stoppable_token` concept checks for the basic interface of a stop token
+    that is copyable and allows polling to see if stop has been requested and
+    also whether a stop request is possible. For a stop token type `T` and a type
+    `CB` that is callable with no arguments, the type `T::callback_type<CB>` shall
+    be well-formed and name the stop callback type to use to register a callback
+    to be executed if a stop request is ever made on a `stoppable_token` of type
+    `T`. The `stoppable_token_for` concept checks for a stop token type compatible
+    with a given callback type. The `unstoppable_token` concept checks for a stop
+    token type that does not allow stopping.
 
 <pre>
-template&lt;typename T>
+template&lt;class T>
   concept stoppable_token =
-    copy_constructible&lt;T> &&
-    move_constructible&lt;T> &&
-    is_nothrow_copy_constructible_v&lt;T> &&
-    is_nothrow_move_constructible_v&lt;T> &&
+    copyable&lt;T> &&
     equality_comparable&lt;T> &&
-    requires (const T& token) {
-      { token.stop_requested() } noexcept -> boolean-testable;
-      { token.stop_possible() } noexcept -> boolean-testable;
+    requires (const T t) {
+      { T{t} } noexcept;
+      { t.stop_requested() } noexcept -> <i>boolean-testable</i>;
+      { t.stop_possible() } noexcept -> <i>boolean-testable</i>;
```

Comment:
I wonder, do we just want to just require that these return `same_as<bool>`?

Was the intent to allow returning `std::true_type` or `std::false_type` as a way of indicating statically that this always returns true/false?

Reply:
I would be for the change.

Reply:
Me too. It's a design change though.

@JeffGarland, @ben-craig: thoughts? Shall we make changes like this that seem uncontroversial and then present LEWG with a roll-up of our design tweaks as a paper in each pre-meeting mailing, or something?

Reply:
I'd be happy with your suggestion of a design tweak roll-up paper.